runtime/vm/lookupmethod.c

```diff
@@ -1031,10 +1031,84 @@ illegalAccessMessage(J9VMThread *currentThread, IDATA badMemberModifier, J9Class
 	char *destModuleMsg = NULL;
 	char packageNameBuf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
 	char *packageNameMsg = NULL;
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	J9Class *unverifiedNestMemberClass = NULL;
+#endif /* J9VM_OPT_VALHALLA_NESTMATES */
 	
 	PORT_ACCESS_FROM_VMC(currentThread);
 	Trc_VM_illegalAccessMessage_Entry(currentThread, J9UTF8_LENGTH(senderClassNameUTF), J9UTF8_DATA(senderClassNameUTF),
 			J9UTF8_LENGTH(targetClassNameUTF), J9UTF8_DATA(targetClassNameUTF), badMemberModifier);
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	/* Values of error codes for nestmates visibility checks on source class is
+	 * odd and on destination/target class is even
+	 */
+	if ((J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS <= errorType)
```

Comment:
This approach feels very hacky.  It will be easy for someone to add a new error message that doesn't follow this pattern without realizing it.

A switch statement will be more explicit and less fragile to future changes:

```
switch(errorType) {
case J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS:
case J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR_SOURCE_CLASS:
case J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_SOURCE_CLASS:
    unverifiedNestMemberClass = targetClass;
    break;

case J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_DEST_CLASS
case J9_VISIBILITY_NEST_HOST_DIFFERENT_PACKAGE_ERROR_DEST_CLASS
case J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_DEST_CLASS
    unverifiedNestMemberClass = senderClass;
    break;
default:
   // Assert unreachable here is good when possible.  In this case, would still need to handle other error cases.
```